addons/block_code/code_generation/blocks_catalog.gd

```diff
@@ -94,6 +94,40 @@ static func _setup_definitions_from_files():
 		_by_class_name[target][block_definition.name] = block_definition
 
 
+static func _add_output_definitions(definitions: Array[BlockDefinition]):
```

Comment:
This is actually for the Area2D and RigidBody2D `on_{entered,exited}` blocks, too. I called them output definitions because they're associated to what the code currently calls `ParameterOutput`. Those are encoded in `display_template` as `[param: TYPE]`. On the other hand, `ParameterInput` is used when the template has `{param: TYPE}`.

Technically, this is the difference between [a parameter and an argument](https://www.educative.io/answers/parameter-vs-argument). What we're trying to handle here are parameters - the things in the function declaration. What we normally use are arguments, the things used in a function call.

So, in the short term it would probably be better to call this `_add_parameter_definitions` and ignore the confusion between `ParameterInput` and `ParameterOutput`. In the longer term it would be better to change `ParameterInput` to `Argument` and `ParameterOutput` to `Parameter`. But I suspect there's a lot of loose "parameter" usage within the code.